src/appleseed/renderer/modeling/postprocessingeffect/postprocessingeffectjob.cpp

```diff
@@ -0,0 +1,120 @@
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Interface header.
+#include "postprocessingeffectjob.h"
+
+// appleseed.renderer headers.
+#include "renderer/modeling/frame/frame.h"
+
+// appleseed.foundation headers.
+#include "foundation/image/canvasproperties.h"
+#include "foundation/image/image.h"
+#include "foundation/image/tile.h"
+#include "foundation/math/ordering.h"
+
+// Standard headers.
+#include <cassert>
+
+using namespace foundation;
+
+namespace renderer
+{
+
+//
+// EffectJob class implementation.
+//
+
+EffectJob::EffectJob(
+    const EffectApplierVector&  effect_appliers,
+    const Frame&                frame,
+    const size_t                tile_x,
+    const size_t                tile_y,
+    const size_t                thread_count)
+  : m_effect_appliers(effect_appliers)
+  , m_frame(frame)
+  , m_tile_x(tile_x)
+  , m_tile_y(tile_y)
+  , m_thread_count(thread_count)
+{
+}
+
+void EffectJob::execute(const size_t thread_index)
+{
+    assert(thread_index < m_effect_appliers.size());
+
+    // Apply the post-processing effect to the tile.
+    m_effect_appliers[thread_index]->apply(
+        m_frame,
+        m_tile_x,
+        m_tile_y);
+}
+
+
+//
+// EffectJobFactory class implementation.
+//
+
+void EffectJobFactory::create(
+    const Frame&                            frame,
+    const EffectJob::EffectApplierVector&   effect_appliers,
+    const size_t                            thread_count,
+    EffectJobVector&                        effect_jobs)
+{
+    // Retrieve frame properties.
+    const CanvasProperties& props = frame.image().properties();
+
+    // Generate tiles in linear order.
+    std::vector<size_t> tiles;
+    linear_ordering(tiles, props.m_tile_count);
+
+    // Make sure the right number of tiles was created.
+    assert(tiles.size() == props.m_tile_count);
+
+    // Create effect jobs, one per tile.
+    for (size_t i = 0; i < props.m_tile_count; ++i)
```

Comment:
`std::` anything you can 
```suggestion
    for (std::size_t i = 0; i < props.m_tile_count; ++i)
```